apps/legacy/tools/history/issue.py

```diff
@@ -139,8 +152,30 @@ def migrate(klass, anon):
             story_set = LogStory.objects.filter(is_duplicate=False)\
                                         .filter(DisplayIssue=i.DisplayIssue)\
                                         .filter(dt__lt=i.dt)
-            if last_i is not None and last_i.IssueID == i.IssueID:
-                story_set = story_set.filter(dt__gte=last_i.dt)
+            if not use_earlier_data:
+                if last_i is not None and last_i.IssueID == i.IssueID:
+                    story_set = story_set.filter(dt__gte=last_i.dt)
+            else:
+                if last_i is not None and last_i.IssueID == i.IssueID:
+                    story_set = story_set.filter(dt__gte=last_i.dt)
+                else:
+                    # issue with stories exists in dump from 2004-08
+                    # do this once per issue
+                    story_existed = OldStory.objects.using('earliest_old_site').filter(\
+                        issue_id=i.IssueID).exists()
+                    #revision_exists = StoryRevision.objects.filter(issue_id=i.IssueID, created__lt=EARLIEST_DATA_DATE).exists()
+                if not story_existed and story_set.exists():
```

Comment:
Since story_existed is only defined on the else branch above, if it goes through the if branch this will raise a NameError 'story_existed not defined'